js/errors.js

```diff
@@ -195,31 +202,55 @@
     ServiceUnavailableError.prototype.constructor = ServiceUnavailableError;
 
 
+    //remove invalid facet filterString from path
+    function removeInvalidFacetFilter(path){
+      var facetFilter = path.slice(path.search('\\*::facets::'), path.search('@'));
+      return path.replace(facetFilter, '');
+    }
     // Errors not associated with http status codes
     // these are errors that we defined to manage errors in the API
     /**
      * @memberof ERMrest
      * @param {string} message error message
+     * @param {string} path path for redirectLink
      * @constructor
      * @desc An invalid facet operator
      */
-    function InvalidFacetOperatorError(message) {
+    function InvalidFacetOperatorError(message, path) {
+
         message = message ? message : module._errorMessage.facetingError;
-        ERMrestError.call(this, '', module._errorStatus.facetingError, message);
+        var redirectPath = removeInvalidFacetFilter(path);
+        ERMrestError.call(this, '', module._errorStatus.facetingError, message, '', redirectPath);
     }
 
     InvalidFacetOperatorError.prototype = Object.create(ERMrestError.prototype);
     InvalidFacetOperatorError.prototype.constructor = InvalidFacetOperatorError;
 
+    // path consits of facet filter alongwith table and schemaName
+    // invalidFilter is removed from the path
+    function removeInvalidFilter(path, invalidFilter){
+      var newPath;
+      if (invalidFilter != ''){
+        newPath = path.slice(path, path.search(invalidFilter));
+      } else{
+        path = path.slice(path);
```

Comment:
Can you explain what are these three lines doing? I cannot exactly understand it.

Reply:
It was meant for cases where a filter is not supplied to error module and the path needs to be sanitized. Earlier `path` used to have `/schema:table/` format with two `/`. Since that has been changed to `schema:table/`, I just need to get everything before filter starts(before `/`). 

e.g. In this case  `isa:dataset/id::eq::269&;`, I should get `isa:dataset` as `Location()` throws only the `path`.
Made this change.